presto-spi/src/main/java/io/prestosql/spi/connector/Name.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spi.connector;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Objects;
+
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+
+public class Name
```

Comment:
Identifier? See https://github.com/prestosql/presto/issues/17 it is all about `identifier`s not `name`s.

Reply:
But we shouldn't confuse with the `Identifer` which is already a part of `AST`

Reply:
@kokosing, at the syntax level they are called identifiers. We didn’t want to overload the term for the metadata APIs to avoid having to use full qualified class names and reduce confusion. In the metadata layer, Name (i.e., the name of an entity such as a table, column, user, role, etc) seems like a reasonable choice.